Update property and methods for better readability.

ADmad opened the pull request:
<!---

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

Always follow the [contribution guidelines](https://github.com/cakephp/cakephp/blob/master/.github/CONTRIBUTING.md) when submitting a pull request. In particular, make sure existing tests still pass, and add tests for all new behavior. When fixing a bug, you may want to add a test to verify the fix.

-->

ADmad committed:
Update property and methods for better readability.

markstory commented on src/Database/Driver.php:
```diff
@@ -229,38 +225,35 @@ public function prepare(Query|string $query): StatementInterface
         }
 
-        return $this->_connection->beginTransaction();
+        return $this->getPDO()->beginTransaction();
```
Should this method be `getPdo()`? I know that PHP isn't case-sensitive but the tooling we use is.

ADmad replied:
Okay, I'll rename it to `getPdo()`.

markstory commented on src/Database/Driver.php:
```diff
@@ -190,16 +189,14 @@ public function version(): string
      * @return \PDO
      */
-    public function getConnection(): PDO
+    public function getPDO(): PDO
```
Should this be public? I think we shouldn't be giving out PDO references that easy as the Driver API could also be wrapping a non-PDO implementation (like in the oracle community driver).

ADmad replied:
Nope, it surely shouldn't be. But I couldn't change it because `Fixture/SchemaLoader` uses the `PDO` instance directly to run the query. So we'll need to figure out how to deal with that and then this method can be made protected.

markstory replied:
Sounds like some goofball found an accidentally public API and used it. :trollface: We could probably fix that usage if we have another way to do an unprepared query. Maybe having a `Driver::exec()` method could be a way forward instead?

ADmad replied:
Done :)

ADmad committed:
Don't expose PDO instance of Driver publicly.

Add Driver::exec() to execute a query directly using PDO instance
without using prepared statements.

markstory commented on src/Database/Driver.php:
```diff
@@ -128,7 +128,7 @@ public function __construct(array $config = [])
      * @return \PDO
      */
-    protected function _connect(string $dsn, array $config): PDO
+    protected function createPDO(string $dsn, array $config): PDO
```
Should this be `createPdo()` to match `getPdo()`?

ADmad replied:
Done

markstory reviewed (approved):
Looks good to me.

ADmad committed:
Rename Driver::createPDO() to Driver::createPdo()
